### PR TITLE
Standalone Banner and Interstitial App Sample

### PR DESCRIPTION
**Pull request by hangxudev**
- [x] Create a new Standalone Project under CriteoSamples Workspace
- [x] Integrate Banner
- [x] Integrate Interstitial

### CODE REVIEW

File: Standalone/Standalone/AdConfigurations.m [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#import "AdConfigurations.h"
+@import CriteoPublisherSdk;
+
+// This is a test Criteo Publisher ID that works for this sample app
+// In your application, update this value with your own Criteo Publisher ID
+static NSString* const criteoPublisherId = @"B-000000";
+
+// Here we use test Criteo Ad Unit IDs in order to return test ads
+// In your application, Criteo Ad Unit IDs are typically the same as your Google Ad Manager's Ad Unit ID
```
Comment (irwansetiawan) [inline Standalone/Standalone/AdConfigurations.m]: We may want to remove this line with `Google Ad Manager` reference.

File: Standalone/Standalone/InterstitialViewController.m [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        [self.interstitial presentFromRootViewController:self];
+        self.displayInterstitialButton.enabled = NO;
+    }
+}
+
+- (void)interstitialDidReceiveAd:(CRInterstitial *)interstitial {
+    self.displayInterstitialButton.enabled = YES;
+    [self.displayInterstitialButton setTitle:@"Display Interstitial" forState:UIControlStateNormal];
+}
+
```
Comment (irwansetiawan) [inline Standalone/Standalone/InterstitialViewController.m]: `interstitialDidReceiveAd:` means bid has been successfully returned, but not necessarily mean we have successfully loaded the creative. 

You may want to use `interstitialIsReadyToPresent:` to notify that an interstitial ad is ready to be displayed.

**Review by irwansetiawan — CHANGES_REQUESTED** [review]

File: Standalone/Standalone/BannerViewController.m [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "BannerViewController.h"
+#import "AdConfigurations.h"
+@import CriteoPublisherSdk;
```
Comment (TurpIF) [inline Standalone/Standalone/BannerViewController.m]: Please use the new preferred way of importing the SDK as described [here](https://publisherdocs.criteotilt.com/app/ios/standalone/)

```
#import <CriteoPublisherSdk/CriteoPublisherSdk.h>
```

File: Standalone/Standalone/AdConfigurations.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+@import CriteoPublisherSdk;
```
Comment (TurpIF) [inline Standalone/Standalone/AdConfigurations.h]: same than below, please use the new preferred way of importing the SDK:

```
#import <CriteoPublisherSdk/CriteoPublisherSdk.h>
```

File: Standalone/Standalone/AdConfigurations.m [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "AdConfigurations.h"
+@import CriteoPublisherSdk;
```
Comment (TurpIF) [inline Standalone/Standalone/AdConfigurations.m]: same than below, please use the new preferred way of importing the SDK:

```
#import <CriteoPublisherSdk/CriteoPublisherSdk.h>
```

File: Standalone/Standalone/AppDelegate.m [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "AppDelegate.h"
+#import "AdConfigurations.h"
+@import CriteoPublisherSdk;
```
Comment (TurpIF) [inline Standalone/Standalone/AppDelegate.m]: same than below, please use the new preferred way of importing the SDK:

```
#import <CriteoPublisherSdk/CriteoPublisherSdk.h>
```

File: Standalone/Standalone/InterstitialViewController.m [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "InterstitialViewController.h"
+#import "AdConfigurations.h"
+@import CriteoPublisherSdk;
```
Comment (TurpIF) [inline Standalone/Standalone/InterstitialViewController.m]: same than for banner, please use the new preferred way of importing the SDK:

```
#import <CriteoPublisherSdk/CriteoPublisherSdk.h>
```

**Review by TurpIF — CHANGES_REQUESTED** [review]

File: Standalone/Standalone/AdConfigurations.m (hunk 1 above)
Comment (hangxudev) [reply Standalone/Standalone/AdConfigurations.m]: Deleted the second line of comment which has `Google Ad Manager`.

File: Standalone/Standalone/InterstitialViewController.m (hunk 2 above)
Comment (hangxudev) [reply Standalone/Standalone/InterstitialViewController.m]: Changed to `interstitialIsReadyToPresent:`

File: Standalone/Standalone/BannerViewController.m (hunk 3 above)
Comment (hangxudev) [reply Standalone/Standalone/BannerViewController.m]: Changed to `#import <CriteoPublisherSdk/CriteoPublisherSdk.h>`

File: Standalone/Standalone/AdConfigurations.h (hunk 4 above)
Comment (hangxudev) [reply Standalone/Standalone/AdConfigurations.h]: Changed to `#import <CriteoPublisherSdk/CriteoPublisherSdk.h>`

File: Standalone/Standalone/AdConfigurations.m (hunk 5 above)
Comment (hangxudev) [reply Standalone/Standalone/AdConfigurations.m]: Changed to `#import <CriteoPublisherSdk/CriteoPublisherSdk.h>`

File: Standalone/Standalone/AppDelegate.m (hunk 6 above)
Comment (hangxudev) [reply Standalone/Standalone/AppDelegate.m]: Changed to `#import <CriteoPublisherSdk/CriteoPublisherSdk.h>`

File: Standalone/Standalone/InterstitialViewController.m (hunk 7 above)
Comment (hangxudev) [reply Standalone/Standalone/InterstitialViewController.m]: Changed to `#import <CriteoPublisherSdk/CriteoPublisherSdk.h>`

**Review by TurpIF — CHANGES_REQUESTED** [review]
Could you please drop the commit https://github.com/criteo/ios-publisher-sdk-examples/pull/9/commits/129fc971fc15a73821727f3dc642243b8b6195e0, this seems unrelated to this PR and would make the history less readable.
If you really need changes from the master (AFAIU you don't), then I invite you to rebase this branch on top of the `master` and push force instead.

**Comment by hangxudev** [comment]
> Could you please drop the commit [129fc97](https://github.com/criteo/ios-publisher-sdk-examples/commit/129fc971fc15a73821727f3dc642243b8b6195e0), this seems unrelated to this PR and would make the history less readable.
> If you really need changes from the master (AFAIU you don't), then I invite you to rebase this branch on top of the `master` and push force instead.

Thank you. I dropped the last two commits(merge and updating Standalone SDK to 3.9), and only keep the Banner and Interstitial commits.
The reason I did the merge is after committing https://github.com/criteo/ios-publisher-sdk-examples/commit/2d83c3851c27c8b476575f9f6789555465d1cd13, the CI failed: https://github.com/criteo/ios-publisher-sdk-examples/runs/953936436?check_suite_focus=true.
It looks like a conflict of `CriteoPublisherSdk` version. I am confused because it was 3.8 for both `AppBidding-GoogleAdManager` and `Standalone`: https://github.com/criteo/ios-publisher-sdk-examples/blob/2d83c3851c27c8b476575f9f6789555465d1cd13/Podfile.
Do you think we should declare a shared `CriteoPublisherSdk` pod for all targets in Podfile or keep it separated under each target?

**Comment by TurpIF** [comment]
> > Could you please drop the commit [129fc97](https://github.com/criteo/ios-publisher-sdk-examples/commit/129fc971fc15a73821727f3dc642243b8b6195e0), this seems unrelated to this PR and would make the history less readable.
> > If you really need changes from the master (AFAIU you don't), then I invite you to rebase this branch on top of the `master` and push force instead.
> 
> Thank you. I dropped the last two commits(merge and updating Standalone SDK to 3.9), and only keep the Banner and Interstitial commits.
> The reason I did the merge is after committing [2d83c38](https://github.com/criteo/ios-publisher-sdk-examples/commit/2d83c3851c27c8b476575f9f6789555465d1cd13), the CI failed: https://github.com/criteo/ios-publisher-sdk-examples/runs/953936436?check_suite_focus=true.
> It looks like a conflict of `CriteoPublisherSdk` version. I am confused because it was 3.8 for both `AppBidding-GoogleAdManager` and `Standalone`: https://github.com/criteo/ios-publisher-sdk-examples/blob/2d83c3851c27c8b476575f9f6789555465d1cd13/Podfile.
> Do you think we should declare a shared `CriteoPublisherSdk` pod for all targets in Podfile or keep it separated under each target?

For know it is good to keep them separate. Even if this is a duplication, this clearly shows how to import the SDK for a dedicated integration. We may take action later if this became too unusable.

**Comment by TurpIF** [comment]
LGTM, could you just solve the conflicting `Podfile.lock` ? I will approve then

**Review by irwansetiawan — DISMISSED** [review]

**Comment by hangxudev** [comment]
I added another commit to upgrade Criteo SDK for Standalone to 3.9. Because after rebased it on the top of master, there were two different versions inside Podfile. And `pod install` returned a similar error like this: https://github.com/criteo/ios-publisher-sdk-examples/runs/953936436?check_suite_focus=true

**Review by TurpIF — APPROVED** [review]

**Pull request merged**